### ISSUE TITLE
[BUG] eslint-lwc engines never runs 

### ISSUE THREAD

**Issue by fredtoutsimplement** [forcedotcom/sfdx-scanner#1151]
**Describe the bug**
I wish that scanner scans my lwc components but what ever I try, this eslint-lwc engines never runs

I played with the config file to ensure that the engine is not disable
I updated my npm, sfdx cli, scanner plugin
I tried many command to target lwc folders 
What ever I do the engine esl-lwc does not run at all.

**To Reproduce**
1- I pull locally on my laptop inteliJ project folder the source code from my Org
2- I run following command  sfdx scanner:run --target ".\force-app\main\default\" --engine "pmd,eslint-lwc,eslint,retire-js,cpd" --category "Design,Best Practices,LWC" --format "csv" --outfile "ResultScan6_nonDFA.csv"

SFDX Connection with the Org is set as readonly because the connection is linked with Prod so I prevent any changes from the IDE targetting the Prod.

**Expected behavior**
I expect that eslint-lwc engines run. But actually only PMD, CPD, and some others are running cf "Screenshot section"

**Screenshots**

PS C:\IntelliJWorkspaces\PROJECT_PROD> sfdx scanner:run --target ".\force-app\main\default\" --engine "pmd,eslint-lwc,eslint,retire-js,cpd" --category "Design,Best Practices,LWC" --format "csv" --outfile "ResultScan6_nonDFA.
csv"
WARNING: We're continually improving Salesforce Code Analyzer. Tell us what you think! Give feedback at https://research.net/r/SalesforceCA
(node:156944) Warning: Deprecated config name: apiVersion. Please use org-api-version instead.
(Use `node --trace-warnings ...` to show where the warning was created)
Executed pmd, found 622 violation(s) across 129 file(s).
Executed retire-js, found 0 violation(s) across 0 file(s).
Executed cpd, found 1964 violation(s) across 35 file(s).
Rule violations were written to ResultScan6_nonDFA.csv.

**Desktop (please complete the following information):**
Software Versions are :
Windows 10
InteliJ Community with plugin Illuminated cloud 2
sfdx-cli/7.209.6 win32-x64 node-v18.15.0
@salesforce/sfdx-scanner 3.15.0

**Additional context**

"Workaround": Have you found any ways to sidestep the problem?
I do not have any workaround I'm stuck

"Urgency": How severe is this problem? "Business stopping" vs "Not business-stopping".
Not business-stopping but audit /code analysis slowed-down

**Comment by fredtoutsimplement** [comment]
Thanks for your support / help

**Comment by jfeingold35** [comment]
@fredtoutsimplement , try running `eslint` and `eslint-lwc` separately, and let us know if that helps.

**Comment by fredtoutsimplement** [comment]
Hello @jfeingold35 ,

Here the command I run **but without success about eslint-lwc**

Command executed : 
PS C:\IntelliJWorkspaces\Project_PROD> sfdx scanner:run --target ".\force-app\main\default\" --engine "pmd,eslint-lwc" --category "Design,Best Practices,LWC" --format "csv" --outfile "ResultScan6_nonDFA.csv"     

Result :             
WARNING: We're continually improving Salesforce Code Analyzer. Tell us what you think! Give feedback at https://research.net/r/SalesforceCA
(node:165644) Warning: Deprecated config name: apiVersion. Please use org-api-version instead.
(Use `node --trace-warnings ...` to show where the warning was created)
Executed pmd, found 646 violation(s) across 135 file(s).
Rule violations were written to ResultScan6_nonDFA.csv.

My config file looks like below ;
{
    "currentVersion": "3.15.0",
    "engines": [
        {
            "name": "pmd",
            "targetPatterns": [
                "**/*.cls",
                "**/*.trigger",
                "**/*.java",
                "**/*.page",
                "**/*.component",
                "**/*.xml",
                "!**/node_modules/**",
                "!**/*-meta.xml"
            ],
            "supportedLanguages": [
                "apex",
                "vf"
            ],
            "disabled": false
        },
        {
            "name": "eslint",
            "targetPatterns": [
                "**/*.js",
                "!**/lwc/**/*.js",
                "!**/node_modules/**",
                "!**/bower_components/**"
            ],
            "disabled": false
        },
        {
            "name": "eslint-lwc",
            "targetPatterns": [
                "**/lwc/**/*.js",
                "!**/node_modules/**"
            ],
            "disabled": false
        },
        {
            "name": "eslint-typescript",
            "targetPatterns": [
                "**/*.ts",
                "!**/node_modules/**",
                "!**/bower_components/**"
            ],
            "disabled": false
        },
        {
            "name": "retire-js",
            "targetPatterns": [],
            "disabled": false
        },
        {
            "name": "cpd",
            "targetPatterns": [
                "**/*.cls",
                "**/*.trigger",
                "**/*.java",
                "**/*.page",
                "**/*.component",
                "**/*.xml",
                "!**/node_modules/**",
                "!**/*-meta.xml"
            ],
            "disabled": false,
            "minimumTokens": 100
        },
        {
            "name": "sfge",
            "targetPatterns": [
                "**/*.cls",
                "**/*.trigger"
            ],
            "disabled": true
        }
    ],
    "javaHome": "C:\\Program Files\\Java\\jdk-11.0.6"
}

**Comment by jfeingold35** [comment]
@fredtoutsimplement , Okay, I think I see what's happening here. You're using `LWC` for the category.
ESLint used to have a `category` property for its rules, which we used for categorization. However, this property has since been removed from ESLint's built-in rules and as such we've stopped using it.
If you look at `sfdx scanner:rule:list --engine eslint,eslint-lwc`, you'll see that ESLint-like rules are "categorized" by their severity ("problem", "suggestion", "layout"). If you use those strings in your `--category` flag instead of "LWC", that should fix the problem.

**Comment by fredtoutsimplement** [comment]
Thanks a lot @jfeingold35 ,

you solved my issue.

I executed the command sfdx scanner:rule:list --engine eslint,eslint-lw

Then I notice that indeed I have to use rules category value dictionary then I executed
PS C:\IntelliJWorkspaces\Project_PROD> sfdx scanner:run --target ".\force-app\main\default\" --engine "pmd,eslint-lwc" --category "Design,Best Practices,problem,suggestion,layout" --format "csv" --outfile "ResultScan6_nonDFA
.csv"

and it works , result below

Executed pmd, found 646 violation(s) across 135 file(s).
Executed eslint-lwc, found 7052 violation(s) across 65 file(s).
Rule violations were written to ResultScan6_nonDFA.csv.


thanks a lot for your support
